### ISSUE TITLE
[C++] Support float16 in writing/reading parquet

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#32728]
Half-float values are not supported in Parquet. Here is a previous issue that talks about that: https://issues.apache.org/jira/browse/PARQUET-1647

So, this will not work:
```java

import pyarrow as pa
import pyarrow.parquet as pq
import numpy as np
arr = pa.array(np.float16([0.1, 2.2, 3]))
table = pa.table({'a': arr})
pq.write_table(table, "test_halffloat.parquet") 
```
`This is a proposal to store float16 values in Parquet as FixedSizeBinary, and then restore them to float16 when reading them back in.`

**Reporter**: [Anja Boskovic](https://issues.apache.org/jira/browse/ARROW-17464) / @anjakefala
**Watchers**: [Rok Mihevc](https://issues.apache.org/jira/browse/ARROW-17464) / @rok
#### Related issues:
- [[Format] HALF precision FLOAT Logical type](https://issues.apache.org/jira/browse/PARQUET-758) (depends upon)
#### PRs and other links:
- [GitHub Pull Request #13947](https://github.com/apache/arrow/pull/13947)

<sub>**Note**: *This issue was originally created as [ARROW-17464](https://issues.apache.org/jira/browse/ARROW-17464). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by anjakefala** [comment]
The [Parquet spec updated](https://github.com/apache/parquet-format/pull/184) is ready for discussion. The next step is to have Java and C++ implementations for float16 ready.

**Comment by assignUser** [comment]
Closed with #37582
